lib/infra_conversion_throttler.rb

```diff
@@ -19,4 +19,43 @@ def self.pending_conversion_jobs
     _log.info("Pending InfraConversionJob: #{pending.count}")
     pending.group_by { |job| job.migration_task.destination_ems }
   end
+
+  # @return [Hash] the list of jobs in state 'running', grouped by conversion host
+  def self.running_conversion_jobs
+    running = InfraConversionJob.where(:state => 'running')
+    _log.info("Running InfraConversionJob: #{running.count}")
+    running.group_by { |job| job.migration_task.conversion_host }
+  end
+
+  # Calculate and apply the limits for all running jobs.
+  # The supported limits are:
+  #   - CPU per conversion host
+  #   - network per conversion host
+  #
+  # The limits can be retrieved per conversion host or from the default setting.
+  # When virt-v2v-wrapper starts the task stores the throttling file path in task.options[:virt-v2v-wrapper]['throttling_file'].
+  # There is no need to adjust limits if virt-v2v-wrapper has not been called by the task.
+  # The resources are evenly split between the jobs on a same conversion host.
+  # There is no need to adjust limits if they have not changed.
+  # Applying the limits is done via the conversion_host which handles the writing.
+  def self.apply_limits
+    running_conversion_jobs.each do |ch, jobs|
+      number_of_jobs = ch.active_tasks.size
+      cpu_limit = ch.cpu_limit || Settings.transformation.limits.cpu_limit_per_host
+      network_limit = ch.network_limit || Settings.transformation.limits.network_limit_per_host
+      jobs.each do |job|
+        migration_task = job.migration_task
+        throttling_file_path = migration_task.options.fetch_path(:virtv2v_wrapper, 'throttling_file')
+        next unless throttling_file_path
+        limits = {
+          :cpu     => cpu_limit == 'unlimited' ? cpu_limit : (cpu_limit.to_i / number_of_jobs).to_s,
```

Comment:
It is minor but this calculation doesn't depend on anything in this loop so you could calculate it [up here](https://github.com/ManageIQ/manageiq/pull/18576/files#diff-8dffd85ef5e1772dce26180aa8ff6d65R44)

Something like
```
cpu_limit = ch.cpu_limit || Settings.transformation.limits.cpu_limit_per_host
cpu_limit = (cpu_limit.to_i / number_of_jobs).to_s unless cpu_limit == "unlimited"
```

Reply:
Moved.